runtime/include/http_session.h

```diff
@@ -187,20 +182,30 @@ http_session_set_response_header(struct http_session *session, int status_code,
 	/* We might not have actually matched a route */
 	if (likely(session->route != NULL)) { http_route_total_increment(&session->route->metrics, status_code); }
 
+	int rc = fputs(http_header_build(status_code), session->response_header.handle);
+	assert(rc != EOF);
+
 	if (status_code == 200) {
-		session->response_header_length = snprintf(session->response_header,
-		                                           HTTP_SESSION_RESPONSE_HEADER_CAPACITY,
-		                                           HTTP_RESPONSE_200_TEMPLATE, content_type, content_length);
-	} else {
-		size_t header_len = http_header_len(status_code);
-		size_t to_copy    = HTTP_SESSION_RESPONSE_HEADER_CAPACITY < header_len
-		                      ? HTTP_SESSION_RESPONSE_HEADER_CAPACITY
-		                      : header_len;
-
-		strncpy(session->response_header, http_header_build(status_code), to_copy);
-		session->response_header_length = to_copy;
+		int rc = auto_buf_flush(&session->response_body);
```

Comment:
shouldn't this be `session->response_header` instead?

Reply:
Actually this should be session->response_body, since we must be flushing it at some point before sending.